app/com/linkedin/drelephant/spark/SparkMetricsAggregator.scala

```diff
@@ -57,21 +57,19 @@ class SparkMetricsAggregator(private val aggregatorConfigurationData: Aggregator
       aggregateResourcesAllocatedMBSeconds(executorInstances, executorMemoryBytes, applicationDurationMillis)
     val resourcesUsedMBSeconds = aggregateResourcesUsedMBSeconds(executorMemoryBytes, totalExecutorTaskTimeMillis)
 
-    val resourcesWastedMBSeconds =
-      ((BigDecimal(resourcesAllocatedMBSeconds) * (1.0 - allocatedMemoryWasteBufferPercentage)) - BigDecimal(resourcesUsedMBSeconds))
-        .toBigInt
-
-   if (resourcesUsedMBSeconds.isValidLong) {
-     hadoopAggregatedData.setResourceUsed(resourcesUsedMBSeconds.toLong)
-    } else {
-      logger.info(s"resourcesUsedMBSeconds exceeds Long.MaxValue: ${resourcesUsedMBSeconds}")
+    val resourcesUsedWithBuffer = resourcesUsedMBSeconds.doubleValue() * (1.0 + allocatedMemoryWasteBufferPercentage)
+    val resourcesWastedMBSeconds =  (resourcesUsedWithBuffer < resourcesAllocatedMBSeconds.doubleValue()) match {
+      case true => resourcesAllocatedMBSeconds.doubleValue() - resourcesUsedWithBuffer
+      case false => 0.0
     }
-
-    if (resourcesWastedMBSeconds.isValidLong) {
-      hadoopAggregatedData.setResourceWasted(resourcesWastedMBSeconds.toLong)
+    //allocated is the total used resource from the cluster.
```

Comment:
I agree it's a little confusing.  I think the best thing is to change hadoopAggregatedData to use resourceAllocated as that right term for what it's expecting. I will create a separate issue and fix it separately after consulting with @nntnag17